### PR TITLE
PythonProfiler: expose `--python-pyspy-process`

### PR DESCRIPTION
**Pull request by roi-granulate**
## Description
Expose py-spy specific flag that forces gProfiler to profile given processes.

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have updated the relevant documentation.
- [ ] I have added tests for new logic.

### CODE REVIEW

File: gprofiler/profilers/python.py [hunk 1]
```diff
@@ -340,6 +341,13 @@ def _should_skip_process(self, process: Process) -> bool:
             action="store_true",
             help="Enable PyPerf in verbose mode (max verbosity)",
         ),
+        ProfilerArgument(
+            name="--python-pyspy-processes",
+            dest="python_pyspy_processes",
+            nargs="+",
+            type=int,
+            help="List of processes (by PID) to profile with py-spy." "This option forces gProfiler to ",
```
Comment (Jongy) [inline gprofiler/profilers/python.py:349]: Fix `help`

File: gprofiler/profilers/python.py [hunk 2]
```diff
@@ -275,7 +277,7 @@ def _select_processes_to_profile(self) -> List[Process]:
             except Exception:
                 logger.exception(f"Couldn't add pid {process.pid} to list")
 
-        return filtered_procs
+        return filtered_procs + [Process(pid) for pid in self._python_pyspy_processes]
```
Comment (Jongy) [inline gprofiler/profilers/python.py:280]: `Process(pid)` will raise if the PID exited, that's okay but let's mention in the `help` that they're expected to be alive forever

File: gprofiler/profilers/python.py [hunk 3]
```diff
@@ -340,6 +342,16 @@ def _should_skip_process(self, process: Process) -> bool:
...
         ),
+        ProfilerArgument(
+            name="--python-pyspy-processes",
+            dest="python_pyspy_processes",
+            nargs="+",
+            type=int,
+            help="List of processes (by PID) to profile with py-spy."
+            "This option forces gProfiler to profile given processes with py-spy, even if"
+            "they are not recognized by gProfiler as Python processes."
+            "Note - gProfiler assumes that the given processes are alive and running forever.",
```
Comment (roi-granulate) [inline gprofiler/profilers/python.py:353]: I don't like the use of "forever" here
@Jongy what other word could fit?

Comment (Jongy) [reply gprofiler/profilers/python.py:353]:
```suggestion
            "Note - gProfiler assumes that the given processes are kept running as long as gProfiler runs.",
```

File: gprofiler/profilers/python.py [hunk 4]
```diff
@@ -340,6 +342,16 @@ def _should_skip_process(self, process: Process) -> bool:
             action="store_true",
             help="Enable PyPerf in verbose mode (max verbosity)",
         ),
+        ProfilerArgument(
+            name="--python-pyspy-processes",
+            dest="python_pyspy_processes",
+            nargs="+",
+            type=int,
+            help="List of processes (by PID) to profile with py-spy."
+            "This option forces gProfiler to profile given processes with py-spy, even if"
```
Comment (Jongy) [inline gprofiler/profilers/python.py:351]: Note that spacing is broken - need to add a space before each line

File: gprofiler/profilers/python.py [hunk 5]
```diff
@@ -340,6 +342,16 @@ def _should_skip_process(self, process: Process) -> bool:
             action="store_true",
             help="Enable PyPerf in verbose mode (max verbosity)",
         ),
+        ProfilerArgument(
+            name="--python-pyspy-processes",
+            dest="python_pyspy_processes",
+            nargs="+",
```
Comment (grariel) [inline gprofiler/profilers/python.py:348]: From argparse docs:
```
'+'. Just like '*', all command-line args present are gathered into a list. Additionally, an error message will be generated if there wasn’t at least one command-line argument present. For example:
```

If I understand correctly, doesn't it require one value?

Comment (roi-granulate) [reply gprofiler/profilers/python.py:348]: It requires one only if the specific argument appears
that is the desired case here I guess (we don't want to support when people specify `--python-pyspy-processes` without PIDs)

Comment (Jongy) [reply gprofiler/profilers/python.py:348]: I think the nicer way is to name the arg singular and use `action="extend"`, so it's provided this way: `--python-pyspy-process xxx --python-pyspy-process yyy`

Comment (grariel) [reply gprofiler/profilers/python.py:348]: Feel free to resolve after finishing with @Jongy's comment

Comment (roi-granulate) [reply gprofiler/profilers/python.py]: I preferred the old way

File: gprofiler/profilers/python.py [hunk 6]
```diff
@@ -275,7 +278,7 @@ def _select_processes_to_profile(self) -> List[Process]:
             except Exception:
                 logger.exception(f"Couldn't add pid {process.pid} to list")
 
-        return filtered_procs
+        return filtered_procs + [Process(pid) for pid in self._python_pyspy_process]
```
[the remark this answers is not in the document]
Comment (roi-granulate) [reply gprofiler/profilers/python.py:281]: We're fine with that, as the `help` mentions - we expect processes to live as long as gProfiler runs.

[the remark this answers is not in the document]
Comment (roi-granulate) [reply gprofiler/profilers/python.py]: Nice

**Review by Jongy — APPROVED** [review]
lgtm

**Review by gAmitFrenkel — APPROVED** [review]

File: gprofiler/profilers/python.py (hunk 5 above)
Comment (roi-granulate) [reply gprofiler/profilers/python.py]: @Jongy I resolve this comment because you approve this PR

**Pull request merged**